src/app/components/GameSetup.tsx

```diff
@@ -1,57 +1,80 @@
-import React from "react";
+import React, { useState } from "react";
 import { Game } from "../page";
-import { FormEvent } from "react";
 
 type Props = {
-  gameSetup: Game;
-  setGameSetup: React.Dispatch<React.SetStateAction<Game>>;
+  gameSetup: Game | undefined;
```

Comment:
gamesetup can be also undefined so we define this as a type union to handle all possible states it may take